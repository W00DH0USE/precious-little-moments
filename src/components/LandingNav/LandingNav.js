import React from 'react';
import { Link } from 'react-router-dom';
import TokenService from '../../services/token-service';
import './LandingNav.css';

function LandingNav(props) {
  return (
    <nav className="Header" style={{ borderBottom: "1px solid #eee" }}>
      <h1>
        <Link to="/">
          PLM
        </Link>
      </h1>

      {!TokenService.hasAuthToken() ?
        <div className="SignOut">
          <div className='Header__not-logged-in'>
            <Link
              to='/login'>
              Login
            </Link>
            <Link
              to='/register'>
              Sign Up
            </Link>
          </div>
        </div>
        :
        <div className="SignOut">
          <div className='Header__logged-in'>
            <Link
              to='/profile'>
              Profile
            </Link>
            <Link
              to='/addPost'>
              Create Memory
            </Link>
            <Link
              onClick={props.handleLogout}
              to='/'>
              Logout
            </Link>
          </div>
        </div>
      }
    </nav>
  )
}

export default LandingNav;