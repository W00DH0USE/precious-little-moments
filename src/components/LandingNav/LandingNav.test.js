import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import LandingNav from './LandingNav';


describe(`LandingNav component`, () => {
    it('renders without crashing', () => {
        const wrapper = shallow(<LandingNav />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});