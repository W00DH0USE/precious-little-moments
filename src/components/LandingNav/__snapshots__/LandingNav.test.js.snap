// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingNav component renders without crashing 1`] = `
<nav
  role="navigation"
>
  <ul>
    <li
      className="title"
    >
      <Link
        to="/"
      >
        SoberFriends
      </Link>
    </li>
    <li
      id="signup"
    >
      Sign Up
    </li>
    <li>
      <Link
        to="/login"
      >
        Posts
      </Link>
    </li>
  </ul>
</nav>
`;
